[Bug]: flink source project pushdown would cause reading all fields to downstream

### What happened?

Select partial fields from arctic table, but there will be all fields without projecting. It will cause unmatched rowtype and real rowdata. And shuffle is using all fields RowType, which may cause ArrayIndexOutOfBoundsException.

### Affects Versions

master/0.3.1

### What engines are you seeing the problem on?

Flink

### How to reproduce

Create a table t with fields a string, b int, c string.  And declare c as partition key. Then select a, c from t. 

### Relevant log output

```shell
java.lang.RuntimeException: 4
	at org.apache.flink.streaming.runtime.io.RecordWriterOutput.pushToRecordWriter(RecordWriterOutput.java:109)
	at org.apache.flink.streaming.runtime.io.RecordWriterOutput.collect(RecordWriterOutput.java:93)
	at org.apache.flink.streaming.runtime.io.RecordWriterOutput.collect(RecordWriterOutput.java:44)
	at org.apache.flink.streaming.runtime.tasks.SourceOperatorStreamTask$AsyncDataOutputToOutput.emitRecord(SourceOperatorStreamTask.java:188)
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:110)
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:101)
	at com.netease.arctic.flink.read.hybrid.reader.ArcticRecordEmitter.emitRecord(ArcticRecordEmitter.java:35)
	at com.netease.arctic.flink.read.hybrid.reader.ArcticRecordEmitter.emitRecord(ArcticRecordEmitter.java:28)
	at org.apache.flink.connector.base.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:128)
	at org.apache.flink.streaming.api.operators.SourceOperator.emitNext(SourceOperator.java:275)
	at org.apache.flink.streaming.runtime.io.StreamTaskSourceInput.emitNext(StreamTaskSourceInput.java:67)
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:65)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:398)
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:191)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:619)
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:583)
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:758)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:573)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 4
	at org.apache.flink.table.data.GenericRowData.isNullAt(GenericRowData.java:129)
	at org.apache.iceberg.flink.RowDataWrapper.get(RowDataWrapper.java:68)
	at org.apache.iceberg.Accessors$PositionAccessor.get(Accessors.java:71)
	at org.apache.iceberg.Accessors$PositionAccessor.get(Accessors.java:58)
	at org.apache.iceberg.PartitionKey.partition(PartitionKey.java:104)
	at com.netease.arctic.flink.shuffle.ShuffleHelper.hashPartitionValue(ShuffleHelper.java:110)
	at com.netease.arctic.flink.shuffle.ReadShuffleRulePolicy$RoundRobinPartitioner.partition(ReadShuffleRulePolicy.java:111)
	at com.netease.arctic.flink.shuffle.ReadShuffleRulePolicy$RoundRobinPartitioner.partition(ReadShuffleRulePolicy.java:83)
	at org.apache.flink.streaming.runtime.partitioner.CustomPartitionerWrapper.selectChannel(CustomPartitionerWrapper.java:54)
	at org.apache.flink.streaming.runtime.partitioner.CustomPartitionerWrapper.selectChannel(CustomPartitionerWrapper.java:33)
	at org.apache.flink.runtime.io.network.api.writer.ChannelSelectorRecordWriter.emit(ChannelSelectorRecordWriter.java:54)
	at org.apache.flink.streaming.runtime.io.RecordWriterOutput.pushToRecordWriter(RecordWriterOutput.java:107)
	... 18 more
```


### Anything else

In #108 supports projection push down. 
But the flinkSchemaRowType in ArcticDynamicSource won't be adjust after projection, which help to build ShuffleHelper.
On the other hand, there is still using all fields schema in buiding RowDataReaderFunction in FlinkSource.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
+++ b/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
@@ -43,14 +43,17 @@ import org.apache.iceberg.Schema;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
 import org.apache.iceberg.util.PropertyUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.netease.arctic.flink.FlinkSchemaUtil.filterWatermark;
 import static com.netease.arctic.table.TableProperties.READ_DISTRIBUTION_HASH_MODE;
@@ -108,13 +111,13 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
     this.arcticTable = arcticTable;
     this.properties = properties;
 
-    readSchema = arcticTable.schema();
     if (projectedSchema == null) {
+      readSchema = arcticTable.schema();
       flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     } else {
-      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
       readSchema = TypeUtil.reassignIds(
           FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema());
+      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
     }
   }
 
@@ -235,6 +238,16 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
 
   @Override
   public void applyProjection(int[][] projectedFields) {
+    int[] projectionFields = new int[projectedFields.length];
+    for (int i = 0; i < projectedFields.length; i++) {
+      org.apache.flink.util.Preconditions.checkArgument(
+          projectedFields[i].length == 1,
+          "Don't support nested projection now.");
+      projectionFields[i] = projectedFields[i][0];
+    }
+    final List<Types.NestedField> columns = readSchema.columns();
+    readSchema = new Schema(Arrays.stream(projectionFields).mapToObj(columns::get).collect(Collectors.toList()));
+    flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     if (arcticDynamicSource instanceof SupportsProjectionPushDown) {
       ((SupportsProjectionPushDown) arcticDynamicSource).applyProjection(projectedFields);
     }

--- a/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -46,6 +46,7 @@ import org.apache.flink.table.types.logical.RowType;
 import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.TypeUtil;
 import org.apache.iceberg.util.PropertyUtil;
 
 import java.util.HashMap;
@@ -151,7 +152,9 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          arcticTable.schema(),
+          projectedSchema == null ?
+              arcticTable.schema() :
+              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -152,9 +152,7 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          projectedSchema == null ?
-              arcticTable.schema() :
-              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
+          scanContext.project(),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
+++ b/flink/v1.12/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
@@ -165,7 +165,7 @@ public class TestKeyed extends FlinkTestBase {
         ")*/ select id, name, op_time_tz, op_time from input");
 
     List<Row> actual =
-        sql("select id, name, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
+        sql("select id, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
             "/*+ OPTIONS(" +
             "'arctic.read.mode'='file'" +
             ", 'streaming'='false'" +
@@ -173,13 +173,13 @@ public class TestKeyed extends FlinkTestBase {
             "");
 
     List<Object[]> expected = new LinkedList<>();
-    expected.add(new Object[]{RowKind.INSERT, 1000004, "a", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000004, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, "d", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, "e", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.INSERT, 1000015, "e", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000015, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
 
     Assert.assertTrue(CollectionUtils.isEqualCollection(DataUtil.toRowList(expected), actual));

--- a/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
+++ b/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
@@ -43,14 +43,17 @@ import org.apache.iceberg.Schema;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
 import org.apache.iceberg.util.PropertyUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.netease.arctic.flink.FlinkSchemaUtil.filterWatermark;
 import static com.netease.arctic.table.TableProperties.READ_DISTRIBUTION_HASH_MODE;
@@ -108,13 +111,13 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
     this.arcticTable = arcticTable;
     this.properties = properties;
 
-    readSchema = arcticTable.schema();
     if (projectedSchema == null) {
+      readSchema = arcticTable.schema();
       flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     } else {
-      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
       readSchema = TypeUtil.reassignIds(
           FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema());
+      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
     }
   }
 
@@ -235,6 +238,16 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
 
   @Override
   public void applyProjection(int[][] projectedFields) {
+    int[] projectionFields = new int[projectedFields.length];
+    for (int i = 0; i < projectedFields.length; i++) {
+      org.apache.flink.util.Preconditions.checkArgument(
+          projectedFields[i].length == 1,
+          "Don't support nested projection now.");
+      projectionFields[i] = projectedFields[i][0];
+    }
+    final List<Types.NestedField> columns = readSchema.columns();
+    readSchema = new Schema(Arrays.stream(projectionFields).mapToObj(columns::get).collect(Collectors.toList()));
+    flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     if (arcticDynamicSource instanceof SupportsProjectionPushDown) {
       ((SupportsProjectionPushDown) arcticDynamicSource).applyProjection(projectedFields);
     }

--- a/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -47,6 +47,7 @@ import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.flink.source.FlinkInputFormat;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.TypeUtil;
 import org.apache.iceberg.util.PropertyUtil;
 
 import java.util.HashMap;
@@ -152,7 +153,9 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          arcticTable.schema(),
+          projectedSchema == null ?
+              arcticTable.schema() :
+              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.14/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -153,9 +153,7 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          projectedSchema == null ?
-              arcticTable.schema() :
-              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
+          scanContext.project(),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.14/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
+++ b/flink/v1.14/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
@@ -169,7 +169,7 @@ public class TestKeyed extends FlinkTestBase {
         ")*/ select id, name, op_time_tz, op_time from input");
 
     List<Row> actual =
-        sql("select id, name, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
+        sql("select id, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
             "/*+ OPTIONS(" +
             "'arctic.read.mode'='file'" +
             ", 'streaming'='false'" +
@@ -177,13 +177,13 @@ public class TestKeyed extends FlinkTestBase {
             "");
 
     List<Object[]> expected = new LinkedList<>();
-    expected.add(new Object[]{RowKind.INSERT, 1000004, "a", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000004, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, "d", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, "e", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.INSERT, 1000015, "e", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000015, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
 
     Assert.assertTrue(CollectionUtils.isEqualCollection(DataUtil.toRowList(expected), actual));

--- a/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
+++ b/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/ArcticDynamicSource.java
@@ -44,14 +44,17 @@ import org.apache.iceberg.Schema;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.types.TypeUtil;
+import org.apache.iceberg.types.Types;
 import org.apache.iceberg.util.PropertyUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 import static com.netease.arctic.flink.FlinkSchemaUtil.filterWatermark;
 import static com.netease.arctic.table.TableProperties.READ_DISTRIBUTION_HASH_MODE;
@@ -109,13 +112,13 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
     this.arcticTable = arcticTable;
     this.properties = properties;
 
-    readSchema = arcticTable.schema();
     if (projectedSchema == null) {
+      readSchema = arcticTable.schema();
       flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     } else {
-      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
       readSchema = TypeUtil.reassignIds(
           FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema());
+      flinkSchemaRowType = (RowType) projectedSchema.toRowDataType().getLogicalType();
     }
   }
 
@@ -239,6 +242,16 @@ public class ArcticDynamicSource implements ScanTableSource, SupportsFilterPushD
 
   @Override
   public void applyProjection(int[][] projectedFields) {
+    int[] projectionFields = new int[projectedFields.length];
+    for (int i = 0; i < projectedFields.length; i++) {
+      org.apache.flink.util.Preconditions.checkArgument(
+          projectedFields[i].length == 1,
+          "Don't support nested projection now.");
+      projectionFields[i] = projectedFields[i][0];
+    }
+    final List<Types.NestedField> columns = readSchema.columns();
+    readSchema = new Schema(Arrays.stream(projectionFields).mapToObj(columns::get).collect(Collectors.toList()));
+    flinkSchemaRowType = FlinkSchemaUtil.convert(readSchema);
     if (arcticDynamicSource instanceof SupportsProjectionPushDown) {
       ((SupportsProjectionPushDown) arcticDynamicSource).applyProjection(projectedFields);
     }

--- a/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -49,6 +49,7 @@ import org.apache.iceberg.expressions.Expression;
 import org.apache.iceberg.flink.FlinkSchemaUtil;
 import org.apache.iceberg.flink.source.FlinkInputFormat;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.TypeUtil;
 import org.apache.iceberg.util.PropertyUtil;
 
 import java.util.HashMap;
@@ -162,7 +163,9 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          arcticTable.schema(),
+          projectedSchema == null ?
+              arcticTable.schema() :
+              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
+++ b/flink/v1.15/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java
@@ -163,9 +163,7 @@ public class FlinkSource {
       RowDataReaderFunction rowDataReaderFunction = new RowDataReaderFunction(
           flinkConf,
           arcticTable.schema(),
-          projectedSchema == null ?
-              arcticTable.schema() :
-              TypeUtil.reassignIds(FlinkSchemaUtil.convert(filterWatermark(projectedSchema)), arcticTable.schema()),
+          scanContext.project(),
           arcticTable.asKeyedTable().primaryKeySpec(),
           scanContext.nameMapping(),
           scanContext.caseSensitive(),

--- a/flink/v1.15/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
+++ b/flink/v1.15/flink/src/test/java/com/netease/arctic/flink/table/TestKeyed.java
@@ -169,7 +169,7 @@ public class TestKeyed extends FlinkTestBase {
         ")*/ select id, name, op_time_tz, op_time from input");
 
     List<Row> actual =
-        sql("select id, name, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
+        sql("select id, op_time, op_time_tz from arcticCatalog." + db + "." + TABLE +
             "/*+ OPTIONS(" +
             "'arctic.read.mode'='file'" +
             ", 'streaming'='false'" +
@@ -177,13 +177,13 @@ public class TestKeyed extends FlinkTestBase {
             "");
 
     List<Object[]> expected = new LinkedList<>();
-    expected.add(new Object[]{RowKind.INSERT, 1000004, "a", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000004, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, "d", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_BEFORE, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, "e", LocalDateTime.parse("2022-06-17T10:11:11.0"),
+    expected.add(new Object[]{RowKind.UPDATE_AFTER, 1000021, LocalDateTime.parse("2022-06-17T10:11:11.0"),
         LocalDateTime.parse("2022-06-17T10:11:11.0").atZone(ZoneId.systemDefault()).toInstant()});
-    expected.add(new Object[]{RowKind.INSERT, 1000015, "e", LocalDateTime.parse("2022-06-17T10:10:11.0"),
+    expected.add(new Object[]{RowKind.INSERT, 1000015, LocalDateTime.parse("2022-06-17T10:10:11.0"),
         LocalDateTime.parse("2022-06-17T10:10:11.0").atZone(ZoneId.systemDefault()).toInstant()});
 
     Assert.assertTrue(CollectionUtils.isEqualCollection(DataUtil.toRowList(expected), actual));

